lib/grape/api/instance.rb

```diff
@@ -79,9 +79,13 @@ def cascade(value = nil)
           end
         end
 
+        def compile!
+          LOCK.synchronize { compile unless instance }
```

Comment:
I think we want to also `return if instance` here to avoid taking the lock every time?

Reply:
Forgot about it, but have now resolved